pipeline-model-definition/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/Utils.java

```diff
@@ -283,152 +264,151 @@ static boolean isRunCausedBy(CpsScript script, String cause, String detail = nul
      * @param detail regarding the cause (currently the user in a UserIdCause)
      * @return if a specific cause must be aborted
      */
-    static boolean shouldRunBeAllowed(Cause causeClass, String cause, String detail){
-        if( causeClass instanceof Cause.UserIdCause && Cause.UserIdCause.simpleName == cause){
-            return detail == null || causeClass.userId == detail
-        }else {
-            return causeClass.class.simpleName.matches("(?i)\\.*${cause}.*")
+    public static boolean shouldRunBeAllowed(Cause causeClass, String cause, String detail) {
+        if (causeClass instanceof Cause.UserIdCause && Cause.UserIdCause.class.getSimpleName().equals(cause)) {
+            return detail == null || ((Cause.UserIdCause) causeClass).getUserId().equals(detail);
+        } else {
+            return causeClass.getClass().getSimpleName().matches("(?i).*" + cause + ".*");
```

Comment:
Unclear what was meant by `\.*cause.*`. This would seem to mean `mycauselala`, `.mycause`, `..`mycausexxx`, etc.? Does not seem to make any sense. I think `.*` (any characters) was meant. Maybe `.*\.`?

Reply:
Called via `#isRunCausedBy` by `TriggeredByConditionalScript`. So seems to support `when { triggeredBy 'SCMTrigger' }` etc.

What the pattern generally seems to do is it allows shorthand, e.g. `triggeredBy 'SCM'` (or `'scm'` thanks to case-insensitivity). I don't see how a `#getSimpleName` would start with `.` though, so all this currently does is effectively `^` (no prefix).

The "fix" appears to change behavior (from prefix to anywhere) so should be done only carefully considered.

Reply:
> all this currently does is effectively `^` (no prefix)

Agreed. Maybe better to just write it that way for now.